extensions/ql-vscode/test/vscode-tests/no-workspace/databases/github-database-prompt.test.ts

```diff
@@ -0,0 +1,267 @@
+    downloadGitHubDatabaseFromUrlSpy = jest
+      .spyOn(databaseFetcher, "downloadGitHubDatabaseFromUrl")
+      .mockResolvedValue(undefined);
+  });
+
+  it("downloads the database", async () => {
+    await promptGitHubDatabaseDownload(
+      octokit,
+      owner,
+      repo,
+      databases,
+      config,
+      databaseManager,
+      storagePath,
+      cliServer,
+      commandManager,
+    );
+
+    expect(downloadGitHubDatabaseFromUrlSpy).toHaveBeenCalledTimes(1);
+    expect(downloadGitHubDatabaseFromUrlSpy).toHaveBeenCalledWith(
+      databases[0].url,
+      databases[0].id,
+      databases[0].created_at,
+      databases[0].commit_oid,
+      owner,
+      repo,
+      octokit,
+      expect.anything(),
+      databaseManager,
+      storagePath,
+      cliServer,
+      true,
+      false,
+    );
+    expect(promptForLanguageSpy).toHaveBeenCalledWith(["swift"], undefined);
+    expect(config.setDownload).not.toHaveBeenCalled();
+  });
+
+  describe("when answering not now to prompt", () => {
+    beforeEach(() => {
+      showNeverAskAgainDialogSpy.mockResolvedValue("Not now");
+    });
+
+    it("does not download the database", async () => {
+      await promptGitHubDatabaseDownload(
+        octokit,
+        owner,
+        repo,
+        databases,
+        config,
+        databaseManager,
+        storagePath,
+        cliServer,
+        commandManager,
+      );
+
+      expect(downloadGitHubDatabaseFromUrlSpy).not.toHaveBeenCalled();
+    });
+  });
+
+  describe("when cancelling prompt", () => {
+    beforeEach(() => {
+      showNeverAskAgainDialogSpy.mockResolvedValue(undefined);
+    });
+
+    it("does not download the database", async () => {
+      await promptGitHubDatabaseDownload(
+        octokit,
+        owner,
+        repo,
+        databases,
+        config,
+        databaseManager,
+        storagePath,
+        cliServer,
+        commandManager,
+      );
+
+      expect(downloadGitHubDatabaseFromUrlSpy).not.toHaveBeenCalled();
+    });
+  });
+
+  describe("when answering never to prompt", () => {
+    beforeEach(() => {
+      showNeverAskAgainDialogSpy.mockResolvedValue("Never");
+    });
+
+    it("does not download the database", async () => {
+      await promptGitHubDatabaseDownload(
+        octokit,
+        owner,
+        repo,
+        databases,
+        config,
+        databaseManager,
+        storagePath,
+        cliServer,
+        commandManager,
+      );
+
+      expect(downloadGitHubDatabaseFromUrlSpy).not.toHaveBeenCalled();
+    });
+
+    it('sets the config to "never"', async () => {
+      await promptGitHubDatabaseDownload(
+        octokit,
+        owner,
+        repo,
+        databases,
+        config,
+        databaseManager,
+        storagePath,
+        cliServer,
+        commandManager,
+      );
+
+      expect(config.setDownload).toHaveBeenCalledTimes(1);
+      expect(config.setDownload).toHaveBeenCalledWith("never");
+    });
```

Comment:
Do you need a test that if the config says "never" then there is no prompt at all?

Reply:
That's quite hard to test because it's in the `GitHubDatabaseModule` which starts an `async` task without awaiting it. I also think it's quite a simple check, so I don't feel like implementing a very complex test for testing this makes a ton of sense.